### ISSUE TITLE
FreeRDP issue on Windows Server 2016 with latest patch update

### ISSUE THREAD

**Issue by AKN1988** [FreeRDP/FreeRDP#6807]
I am using freerdp to connect to a Windows Server 2016 machine. It was working fine before, but after certain security patch updates on the machine, I am no longer able to take remote connection. The connection works through mstsc but not through freerdp. 
Free RDP version used : 2.2.0
Environment : Windows Server 2016
Patches applied : kb4593226, kb4535680, kb4598243, kb4593226, kb4535680, kb4598243, kb4593226, kb4535680, kb4593226, kb4535680, kb4598243, kb4535680, kb4598243, kb4593226, kb4535680, kb4598243, kb4593226, kb4535680, kb4598243, kb4593226, kb4535680, kb4598243

Does any of these patches have any breaking changes related to FreeRDP?
Kindly help.

**Comment by akallabeth** [comment]
@AKN1988 well, could you add some log output from freerdp and also check if it works with nightly/master ?
We don´t have seen any such issue until now, but who knows ^^

**Comment by akallabeth** [comment]
Just verified, connection to a server 2016 machine (not in a domain) works on our end (latest patch level)

**Comment by surrocat** [comment]
Hi, I can't connect to Windows 2016 either, tried the following form on Ubuntu 20.10, with a server that works using Windows rdp:

xfreerdp /u:CONTOSO\JohnDoe /p:Pwd123! /v:rdp.contoso.com

and it fails, saying "failed to connect to rdp.contoso.com"

Can you give a typical command line for a Windows 2016 connection that works? Thanks

**Comment by akallabeth** [comment]
@surrocat you just have one. the error sounds more like your DNS can not resolve your host? but without a log that is hard to tell.

**Comment by surrocat** [comment]
I just tried with the actual IP address of the Windows server instead of 'rdp.contoso.com' and got exactly the same error.

**Comment by surrocat** [comment]
PS the Windows server has DTLS enabled. On Windows I had to install KB2574819 and KB259268 in 2018 to get rdp to connect.

**Comment by akallabeth** [comment]
@surrocat I´m still tapping in the dark, could you add some log (as requested by the issue template and, just above myself)? Your description is too generic to give you specific answers.

**Comment by kunal7898** [comment]
> I just tried with the actual IP address of the Windows server instead of 'rdp.contoso.com' and got exactly the same error.

Is there a way to add logs? If yes can you share some sample and that will be helpfull

**Comment by surrocat** [comment]
This was the output:
ben@ben-desktop:~$ xfreerdp /u:CONTOSO\JohnDoe /p:Pwd123! /v:xxx.xx.xxx.xx
[12:42:12:003] [2633:2634] [INFO][com.freerdp.core] - freerdp_connect:freerdp_set_last_error_ex resetting error state
[12:42:12:004] [2633:2634] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpdr
[12:42:12:007] [2633:2634] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpsnd
[12:42:12:007] [2633:2634] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[12:42:13:275] [2633:2634] [INFO][com.freerdp.primitives] - primitives autodetect, using generic
[12:42:13:286] [2633:2634] [INFO][com.freerdp.core] - freerdp_tcp_is_hostname_resolvable:freerdp_set_last_error_ex resetting error state
[12:42:13:287] [2633:2634] [INFO][com.freerdp.core] - freerdp_tcp_connect:freerdp_set_last_error_ex resetting error state
[12:42:28:302] [2633:2634] [ERROR][com.freerdp.core] - freerdp_tcp_connect:freerdp_set_last_error_ex ERRCONNECT_CONNECT_FAILED [0x00020006]
[12:42:28:302] [2633:2634] [ERROR][com.freerdp.core] - failed to connect to xxx.xx.xxx.xx
ben@ben-desktop:~$ 

It reminds me of the error Windows 7 gave in 2018 (see jpg) which is why I suspect problems with DTSL (does FreeRDP support it?

**Comment by akallabeth** [comment]
@surrocat `DTLS` you mean? That is only for extensions (`UDP` transport for certain `RDP` channels) which we do not support. But what you see here happens way earlier.
This is, as suspected, an issue that the server is not listening on `3389`.
btw, which jpeg do you mean?

**Comment by surrocat** [comment]
![rdpW7](https://user-images.githubusercontent.com/78930998/107772387-18660c00-6d34-11eb-8735-016652e6a9ab.jpg)

**Comment by surrocat** [comment]
DTLS yes, Windows 7 also said it can't connect but it was solved by installing the KB's mentioned.

**Comment by akallabeth** [comment]
@surrocat this is a server configuration issue then, sorry.

**Comment by akallabeth** [comment]
@surrocat simple test, check if TCP port `3389` is open on the server

**Comment by surrocat** [comment]
The solution in 2018 did not involve changes in the server, only updating the Windows 7 client by KB2574819 and KB259268. Will try to ask about ports...

**Comment by akallabeth** [comment]
https://support.microsoft.com/en-us/topic/an-update-is-available-that-adds-support-for-dtls-in-windows-7-sp1-and-windows-server-2008-r2-sp1-2c0e25cd-e8ff-968b-4aff-3729a3a5f7db
explains it a bit. If the clients support `UDP` then this is required for windows 7, yes.
But disabling `TCP` will break lots of stuff (think frewalls/NAT/... that do not like to forward UDP traffic for random applications)
From the log I infer that exactly that is the case for your server.

**Comment by surrocat** [comment]
Ok thanks. Unfortunately, as a user/client I have no influence on the server side, so I am stuck... 

It means that FreeRDP and Windows RDP have diverged in 2018 and are not compatible.

**Comment by akallabeth** [comment]
@surrocat your conclusion is a bit misleading.
FreeRDP and windows RDP are perfectly compatible, but your server is configured in a non compatible way.
(it uses an extension that clients `may` support and deactivates the common case most clients use or need to fall back on when the connection does not support `UDP`)

**Comment by surrocat** [comment]
I understand that introducing udp may not be optimal from our point of view. But Microsoft have decided to implement it and most IT departments probably just follow their changes in setting up the server. I had no luck in convincing them to go back to the previous way of doing things.

**Comment by akallabeth** [comment]
@surrocat no, that is not the point. not seen any server configured as strangely as yours to be honest, because that feature is `optional` and can be enabled without shutting out all other clients. You need to walk the extra mile disabling the traditional way to achieve that ;)

**Comment by surrocat** [comment]
@akallabeth The server has been set up like this since 2018 following a Windows update, and on my side the updates made Windows 7 connect immediately (no changes to Firewall required). It has been running smoothly ever since. I don't think it is an exceptional configuration.

Can you explain what is the crucial setting on the server side that changed in 2018? 

There must be many more FreeRDP users having that problem (maybe without knowing the reason), and it seems to have been requested before https://github.com/FreeRDP/FreeRDP/issues/4978, so I'm asking if we can please have a "Windows Compatibility" option? 

Certainly it would help in order to make FreeRDP/Ubuntu more versatile and widely adopted. Many thanks for looking into this!

**Comment by akallabeth** [comment]
@surrocat you seem to misunderstand.
The transport is either `TCP Only` or `TCP or UDP`.
So, if `UDP` is successful that is used, otherwise `TCP` is used.

There is no crucial setting as all our test VM with server 2016, 2019, ... run quite fine (even with `TCP or UDP`  enabled)

UDP uses different ports though, so maybe that is the problem with your server. (in fact, I´ve never heard nor seen a problem such as yours)

The fact that you needed to update your windows 7 clients confirms my suspicion, as that should not have been necessary. (only the `NLA` hotfix a few years back did break authentication on `strict` configured servers)
The updates for windows 7 are only required to enhance speed and support more modern drawing, not to allow connection at all

So, in short, this has nothing to do with `FreeRDP` not being compatible but some configuration on your server. (all the docs say that `TCP 3389` is still the main transport for `RDP`)

Of course, `UDP` transport (as in #4978 ) would be a nice to have thing, but we´re currently busy with more important stuff than some optional enhancement. (but we´ll always accept pull requests ;))

Only setting I know of is this one: https://admx.help/?Category=Windows_10_2016&Policy=Microsoft.Policies.TerminalServer::TS_SELECT_TRANSPORT
